analyses/filter-mutation-frequencies-tables/filter-mutation-frequencies-tables.Rmd

```diff
@@ -0,0 +1,309 @@
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+
+Set up directories. 
+```{r}
+# directories for input and output files
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+analyses_dir <- file.path(root_dir, "analyses")
+module_dir <- file.path(analyses_dir, "filter-mutation-frequencies-tables")
+snv_results_dir <- file.path(analyses_dir,  "snv-frequencies/results")
+cnv_results_dir <- file.path(analyses_dir,  "cnv-frequencies/results")
+fusion_results_dir <- file.path(analyses_dir,  "fusion-frequencies/results")
+results_dir <- file.path(module_dir, "results")
+
+# Create results folder if it doesn't exist
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+```
+
+## Get Ensembl gene identifiers
+
+Load GenCode gtf file
+```{r}
+# read file
+gtf <- file.path(data_dir, "gencode.v38.primary_assembly.annotation.gtf.gz")
+gencode_gtf <- rtracklayer::import(con = gtf) %>% as.data.frame() %>% tibble()
+```
+
+Get Ensembl identifiers
+```{r}
+# filter for gene rows and get ids
+ensg_ids <- gencode_gtf %>% 
+  dplyr::filter(type == "gene") %>% 
+  dplyr::mutate(gene_id = stringr::str_extract(gene_id, "ENSG\\d+")) %>% 
+  pull(gene_id) %>% 
+  unique() 
+ensg_ids <- ensg_ids[!is.na(ensg_ids)]
+
+# Remove gencode dataframe
+rm(gene_code_annot)
+```
+
+Get OpenPedCan sample IDs (Kids_First_Biospecimen_ID)
+```{r}
+sample_ids <- readr::read_tsv(file.path(data_dir, "histologies.tsv"), 
+                              guess_max = 10000, show_col_types = FALSE) %>% 
+  dplyr::filter(sample_type == "Tumor") %>% 
+  dplyr::select(Kids_First_Biospecimen_ID, cohort)
+```
+
+## Filter mutation frequencies tables
+
+Function to filter for required Ensembl gene identifiers and write to file
+```{r}
+
+filter_freq_table <- function(freq_file, ensg_ids) {
+  
+  # Read gene level table 
+  freq_table <-  readr::read_tsv(freq_file, guess_max = 10000, 
+                                 show_col_types = FALSE) 
+  
+  # Write filtered TSV file
+  file_name <- basename(freq_file)
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+  readr::write_tsv(file.path(results_dir, file_name))
+
+  # Write filtered JSON file
+  json_file <- paste(unlist(str_split(file_name, "\\."))[1], "json", sep = ".")
+  freq_table %>% dplyr::filter(targetFromSourceId %in% ensg_ids) %>% 
+    jsonlite::write_json(file.path(results_dir,  json_file))  
+
+return(freq_table)
+}
+```
+
+#### Gene level SNV frequencies
+```{r}
+# Remove Ensembl gene identifiers
+gene_level_snv <- filter_freq_table(
+  file.path(snv_results_dir, 
+            "gene-level-snv-consensus-annotated-mut-freq.tsv.gz"), ensg_ids) 
+
+# Ensembl gene identifiers removed
+filtered_ensg <- gene_level_snv %>% 
+  dplyr::filter(!targetFromSourceId %in% ensg_ids) %>% 
+  dplyr::select(Gene_symbol, targetFromSourceId) %>%
+  dplyr::rename(Ensembl_ID = targetFromSourceId) %>% 
+  dplyr::distinct()
+
```

Comment:
For all tables - could you please print out `filtered_ensg` here so that we can see the ENSG IDs being filtered?